File: cowork-admin/src/main/java/com/shiroyk/cowork/coworkadmin/repository/user/UserRepository.java
package com.shiroyk.cowork.coworkadmin.repository.user;

import com.shiroyk.cowork.coworkcommon.model.user.User;
import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface UserRepository extends MongoRepository<User, String> {
    Optional<User> findUserByUsername(String name);
}

File: cowork-user/service/src/main/kotlin/user/service/repository/UserRepository.kt
package user.service.repository

import org.springframework.data.mongodb.repository.MongoRepository
import org.springframework.stereotype.Repository
import user.service.entity.UserEntity
import java.util.*

@Repository
interface UserRepository : MongoRepository<UserEntity, String> {

    fun findByIdIn(ids: List<String>): List<UserEntity>

    fun findByUsername(username: String): Optional<UserEntity>

    fun findByEmail(email: String): Optional<UserEntity>
}